packages/tinytest/README.md

```diff
@@ -1,26 +1,314 @@
-This is an internal Meteor package.
-
-API:
-test.ok(doc)
-test.expect_fail()
-test.fail(doc)
-test.exception(exception)
-test.runId()
-test.equal(actual, expected, message, not)
-test.notEqual(actual, expected, message)
-test.instanceOf(obj, klass, message)
-test.notInstanceOf(obj, klass, message)
-test.matches(actual, regexp, message)
-test.notMatches(actual, regexp, message)
-test.throws(func)
-test.isTrue(v, msg)
-test.isFalse(v, msg)
-test.isNull(v. msg)
-test.isNotNull(v, msg)
-test.isUndefined(v, msg)
-test.isNotUndefined(v, msg)
-test.isNaN(v, msg)
-test.isNotNaN(v, msg)
-test.include(s, v, message, not)
-test.notInclude(s, v, message)
-test.length(obj, expected_length, msg)
\ No newline at end of file
+# TinyTest
+
+After reading this, you'll know:
+
+1. What Tinytest is used for.
+1. How to include tests in your package.
+1. How to structure your test file(s).
+1. When to use synchronous or asynchronous tests.
+1. How to run your tests.
+1. What assertions and utility methods are available to you.
+
+## What is Tinytest?
+
+Tinytest is the official Meteor package test runner.
+
+It's used to test *local* packages. You cannot test non-local packages (for example packages you have included using `meteor add some:package-name`) unless you clone their repos and make them available locally. The assumption is, of course, that you are testing packages under development.
+
+Tinytest is specifically designed for *unit testing*. From [wikipedia](https://en.wikipedia.org/wiki/Unit_testing):
+
+> ... Intuitively, one can view a unit as the smallest testable part of an application. In procedural programming, a unit could be an entire module, but it is more commonly an individual function or procedure. In object-oriented programming, a unit is often an entire interface, such as a class, but could be an individual method.
+
+> ... Ideally, each test case is independent from the others. Substitutes such as method stubs, mock objects, fakes, and test harnesses can be used to assist testing a module in isolation.
+
+The object of a unit test is to prove functionality you have written. It is not to prove third-party or other package or library code which may form a part of your codebase. This may require that you head up your tests with *stubs, mocks/spies* or *fakes* which satisfy the demands of your code, but are non-functional (or have limited functionality). Alternatively, instead of placing the stubs in your test file, you could put them into a separate `stubs.js` file, which you `api.addFiles`, or even create and add a stubs package which you then `api.use`.
+
+From [martinfowler.com](http://www.martinfowler.com/articles/mocksArentStubs.html):
+
+> Fake objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).
```

Comment:
Minimongo is a good example that the last example here is not correct or rather out of date.